test/e2e_test.go

```diff
@@ -468,6 +515,69 @@ func consumeMemory(client clientset.Interface, podName string) error {
 	return watchPodReadyStatus(client, metav1.NamespaceDefault, podName, currentPod.ResourceVersion)
 }
 
+func consumeWithInitContainerCPU(client clientset.Interface, podName string) error {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: podName},
+		Spec: corev1.PodSpec{Containers: []corev1.Container{
+			{
+				Name:    podName,
+				Command: []string{"./consume-cpu/consume-cpu"},
+				Args:    []string{"--duration-sec=60", "--millicores=50"},
+				Image:   "gcr.io/kubernetes-e2e-test-images/resource-consumer:1.5",
+				Resources: corev1.ResourceRequirements{
+					Requests: map[corev1.ResourceName]resource.Quantity{
+						corev1.ResourceCPU: mustQuantity("100m"),
+					},
+				},
+			},
+		},
+			InitContainers: []corev1.Container{
+				{
+					Name:    "init-container",
+					Command: []string{"./consume-cpu/consume-cpu"},
+					Image:   "gcr.io/kubernetes-e2e-test-images/resource-consumer:1.5",
+				},
+			}},
+	}
+
+	currentPod, err := client.CoreV1().Pods(metav1.NamespaceDefault).Create(context.TODO(), pod, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	return watchPodReadyStatus(client, metav1.NamespaceDefault, podName, currentPod.ResourceVersion)
```

Comment:
Not 100% sure if we should wait for pod ready here. Would be good to check what metrics are reported when init container is running.

Reply:
when init container is running, the init container reports the pod metrics.
After init container terminated, the init container reports zero pod metrics ephemerally.
Afer that mertrics-server  can get business container metrics normally